### DISCUSSION TITLE
Problems on LDAP connection

### DISCUSSION THREAD

**Discussion by cedgnans** [opendcim/openDCIM#1578] [Q&A]
![image](https://github.com/user-attachments/assets/28434ee5-81be-47fd-a688-52a57857818b)

/var/log/apache2$ sudo tail -f error.log

[Fri Nov 15 18:21:54.346685 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP authentication successful, but access denied based on lacking group membership.  Username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:21:54.346738 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP Authentication failed for username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.140716 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] User search filter: (|(uid=noc)(sAMAccountName=noc)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149330 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP authentication successful, but access denied based on lacking group membership.  Username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149387 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP Authentication failed for username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.463098 2024] [php:notice] [pid 15591] [client 192.168.60.199:40890] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466642 2024] [php:warn] [pid 15591] [client 192.168.60.199:40890] PHP Warning:  ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466753 2024] [php:error] [pid 15591] [client 192.168.60.199:40890] PHP Fatal error:  Uncaught TypeError: ldap_get_entries(): Argument #2 ($result) must be of type LDAP\\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n  thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:55:42.121754 2024] [php:notice] [pid 15557] [client 192.168.60.199:50826] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125428 2024] [php:warn] [pid 15557] [client 192.168.60.199:50826] PHP Warning:  ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125514 2024] [php:error] [pid 15557] [client 192.168.60.199:50826] PHP Fatal error:  Uncaught TypeError: ldap_get_entries(): Argument #2 ($result) must be of type LDAP\\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n  thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php?logout

I'm trying to connect my LDAP server to openDCIM. The connection is successful as the log shows below but when I enter the user authentication information which I've configured in LDAP, it doesn't automatically redirect me to the dashboard.
I had to delete the “login_ldap.php” part manually in the url to access the following interface

![image](https://github.com/user-attachments/assets/71e2eb63-459b-4340-9562-5d2db57c381a)

Any idea on how it can redirect automatically?

**Comment by wilpig** [comment]
You're missing a PHP module. Follow the installation instructions in the wiki.

PHP Warning: ldap_read()

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: cedgnans ***@***.***>
Sent: Friday, November 15, 2024 12:07:07 PM
To: opendcim/openDCIM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [opendcim/openDCIM] Problems on LDAP connection (Discussion #1578)


image.png (view on web)<https://github.com/user-attachments/assets/28434ee5-81be-47fd-a688-52a57857818b>

/var/log/apache2$ sudo tail -f error.log

[Fri Nov 15 18:21:54.346685 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP authentication successful, but access denied based on lacking group membership. Username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:21:54.346738 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP Authentication failed for username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.140716 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] User search filter: (|(uid=noc)(sAMAccountName=noc)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149330 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP authentication successful, but access denied based on lacking group membership. Username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149387 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP Authentication failed for username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.463098 2024] [php:notice] [pid 15591] [client 192.168.60.199:40890] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466642 2024] [php:warn] [pid 15591] [client 192.168.60.199:40890] PHP Warning: ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466753 2024] [php:error] [pid 15591] [client 192.168.60.199:40890] PHP Fatal error: Uncaught TypeError: ldap_get_entries(): Argument #2<https://github.com/opendcim/openDCIM/issues/2> ($result) must be of type LDAP\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:55:42.121754 2024] [php:notice] [pid 15557] [client 192.168.60.199:50826] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125428 2024] [php:warn] [pid 15557] [client 192.168.60.199:50826] PHP Warning: ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125514 2024] [php:error] [pid 15557] [client 192.168.60.199:50826] PHP Fatal error: Uncaught TypeError: ldap_get_entries(): Argument #2<https://github.com/opendcim/openDCIM/issues/2> ($result) must be of type LDAP\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php?logout

I'm trying to connect my LDAP server to openDCIM. The connection is successful as the log shows below but when I enter the user authentication information which I've configured in LDAP, it doesn't automatically redirect me to the dashboard.
I had to delete the “login_ldap.php” part manually in the url to access the following interface

image.png (view on web)<https://github.com/user-attachments/assets/71e2eb63-459b-4340-9562-5d2db57c381a>

Any idea on how it can redirect automatically?

—
Reply to this email directly, view it on GitHub<https://github.com/opendcim/openDCIM/discussions/1578>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAMTY6X7CUPEYQESSP23EWT2AYZ4XAVCNFSM6AAAAABR3WHYT2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ4TAMZYHE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/opendcim/openDCIM/discussions/1578", "url": "https://github.com/opendcim/openDCIM/discussions/1578", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by wilpig** [comment]
You also have a group membership problem too LDAP authentication successful, but access denied based on lacking group membership.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: cedgnans ***@***.***>
Sent: Friday, November 15, 2024 12:07:07 PM
To: opendcim/openDCIM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [opendcim/openDCIM] Problems on LDAP connection (Discussion #1578)


image.png (view on web)<https://github.com/user-attachments/assets/28434ee5-81be-47fd-a688-52a57857818b>

/var/log/apache2$ sudo tail -f error.log

[Fri Nov 15 18:21:54.346685 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP authentication successful, but access denied based on lacking group membership. Username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:21:54.346738 2024] [php:notice] [pid 15591] [client 192.168.60.199:48766] LDAP Authentication failed for username: dcim, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.140716 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] User search filter: (|(uid=noc)(sAMAccountName=noc)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149330 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP authentication successful, but access denied based on lacking group membership. Username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:22:19.149387 2024] [php:notice] [pid 15600] [client 192.168.60.199:53510] LDAP Authentication failed for username: noc, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.463098 2024] [php:notice] [pid 15591] [client 192.168.60.199:40890] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466642 2024] [php:warn] [pid 15591] [client 192.168.60.199:40890] PHP Warning: ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:25:47.466753 2024] [php:error] [pid 15591] [client 192.168.60.199:40890] PHP Fatal error: Uncaught TypeError: ldap_get_entries(): Argument #2<https://github.com/opendcim/openDCIM/issues/2> ($result) must be of type LDAP\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php
[Fri Nov 15 18:55:42.121754 2024] [php:notice] [pid 15557] [client 192.168.60.199:50826] User search filter: (|(uid=dcim)(sAMAccountName=dcim)), referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125428 2024] [php:warn] [pid 15557] [client 192.168.60.199:50826] PHP Warning: ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php?logout
[Fri Nov 15 18:55:42.125514 2024] [php:error] [pid 15557] [client 192.168.60.199:50826] PHP Fatal error: Uncaught TypeError: ldap_get_entries(): Argument #2<https://github.com/opendcim/openDCIM/issues/2> ($result) must be of type LDAP\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 52, referer: https://10.225.5.209/login_ldap.php?logout

I'm trying to connect my LDAP server to openDCIM. The connection is successful as the log shows below but when I enter the user authentication information which I've configured in LDAP, it doesn't automatically redirect me to the dashboard.
I had to delete the “login_ldap.php” part manually in the url to access the following interface

image.png (view on web)<https://github.com/user-attachments/assets/71e2eb63-459b-4340-9562-5d2db57c381a>

Any idea on how it can redirect automatically?

—
Reply to this email directly, view it on GitHub<https://github.com/opendcim/openDCIM/discussions/1578>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAMTY6X7CUPEYQESSP23EWT2AYZ4XAVCNFSM6AAAAABR3WHYT2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ4TAMZYHE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/opendcim/openDCIM/discussions/1578", "url": "https://github.com/opendcim/openDCIM/discussions/1578", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by cedgnans** [comment]
I've followed exactly what is in the wiki to install OPENDCIM.
I don't know what I have really missed something cause I did it twice on
two different servers and got the same problem

Le ven. 15 nov. 2024 à 19:11, Wilbur Longwisch ***@***.***> a
écrit :

**Comment by cedgnans** [comment]
I don't really get the group membership problem that you are taking about

Le ven. 15 nov. 2024 à 19:13, Wilbur Longwisch ***@***.***> a
écrit :

**Comment by wilpig** [comment]
Your user isn't in the group defined for access according to the error or you have the group defined wrong entirely so it isn't getting a valid lookup.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by wilpig** [comment]
Check your ldap parameters as well because with a user name of noc I'd not expect the filter to be looking for uid=dcim typically that'd be uid=noc

LDAP Authentication failed for username: noc,
User search filter: (|(uid=dcim)(sAMAccountName=dcim))

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by cedgnans** [comment]
I'm using openldap so the sAMAccountName should not be the one I should
choose right?

Le ven. 15 nov. 2024 à 19:26, Wilbur Longwisch ***@***.***> a
écrit :

**Comment by cedgnans** [comment]
So if I summarize what you said.
1) There is a PHP extension that is missing and I should review the wiki
page
2)I should check my LDAP parameter as well on OpenDCIM web interface

But I don't really understand that part
"Your user isn't in the group defined for access according to the error or
you have the group defined wrong entirely so it isn't getting a valid
lookup.
"

Le ven. 15 nov. 2024 à 20:45, Cedric Brian ***@***.***> a
écrit :

**Reply by wilpig** [reply]
This is how the groups are set up on the dev site.  What do your groups look like?  show the results of an ldap search for your groups and verify that your user is listed.

<img width="1273" alt="Screenshot 2024-11-15 at 1 55 59 PM" src="https://github.com/user-attachments/assets/fedaad4a-1104-43f6-be48-aa77924c9f8e">

You also need to look at the ldap set up portion because it looks like you have dcim hardcoded in as the username instead of doing a lookup.  Again see how the dev site is set up as an example.

<img width="1277" alt="Screenshot 2024-11-15 at 1 59 00 PM" src="https://github.com/user-attachments/assets/5f6a1784-6a79-462b-8f90-3c63026def96">

**Reply by cedgnans** [reply]
![image](https://github.com/user-attachments/assets/6972b46c-c9a6-48e2-93d6-9190860b27d7)
![image](https://github.com/user-attachments/assets/b22a5edb-a28d-4a49-8def-090be1f60390)

**Reply by wilpig** [reply]
Your bind dn has the user dcim hard coded.  see the example out to the right.  It isn't using a single bind to do lookups it is passing the current user in and verifying.

**Reply by wilpig** [reply]
Looking at your group mappings there it looks like you have the site access set to a context and not a group as well.  Based on the pathing of your other groups there is no way that site access is valid.

**Reply by cedgnans** [reply]
> Your bind dn has the user dcim hard coded. see the example out to the right. It isn't using a single bind to do lookups it is passing the current user in and verifying.

Yeah, It was just for test reason. I've forgotten to change it

**Reply by cedgnans** [reply]
> Looking at your group mappings there it looks like you have the site access set to a context and not a group as well. Based on the pathing of your other groups there is no way that site access is valid.

However, access to the site is valid. I don't really understand

**Reply by cedgnans** [reply]
![image](https://github.com/user-attachments/assets/fc1a98dd-0871-4572-b8eb-5b6d0d047c6d)


I still having the redirection  problem and the logs look like
[Mon Nov 18 15:42:55.456287 2024] [php:notice] [pid 46536] [client 192.168.60.199:58274] User search filter: (uid=noc), referer: https://10.225.5.209/login_ldap.php?logout
[Mon Nov 18 15:42:55.458112 2024] [php:notice] [pid 46536] [client 192.168.60.199:58274] noc has ReadAccess by membership in cn=ReadAccess,cn=openDCIM,ou=groups,dc=trinet,dc=bj, referer: https://10.225.5.209/login_ldap.php?logout
[Mon Nov 18 15:42:55.460157 2024] [php:warn] [pid 46536] [client 192.168.60.199:58274] PHP Warning:  ldap_read(): Search: No such object in /opt/openDCIM/openDCIM-23.04/login_ldap.php on line 50, referer: https://10.225.5.209/login_ldap.php?logout
[Mon Nov 18 15:42:55.460231 2024] [php:error] [pid 46536] [client 192.168.60.199:58274] PHP Fatal error:  Uncaught TypeError: ldap_get_entries(): Argument #2 ($result) must be of type LDAP\\Result, false given in /opt/openDCIM/openDCIM-23.04/login_ldap.php:52\nStack trace:\n#0 /opt/openDCIM/openDCIM-23.04/login_ldap.php(52): ldap_get_entries()\n#1 /opt/openDCIM/openDCIM-23.04/login_ldap.php(191): isUserInLDAPGroup()\n#2 {main}\n  thrown in /opt/openDCIM/openDCIM-23.04/login_ldap.php on 
line 52, referer: https://10.225.5.209/login_ldap.php?logout


And I've just realized that users noc and customer don't have the correct interface ( They are not able to do anything)

This is for NOC
![image](https://github.com/user-attachments/assets/9e165eab-3e8d-4f24-9fd7-8f9a067b928f)

This is for customer
![image](https://github.com/user-attachments/assets/29027ade-441a-46fd-9a49-f6bc78f4346d)

**Reply by wilpig** [reply]
One of your ldap groups is wrong or missing.

https://github.com/opendcim/openDCIM/blob/master/login_ldap.php#L50  if you didn't fix that site access group then i'd start there.

**Reply by cedgnans** [reply]
You were right. Thank you very much. I've forgotten about the BulkOperations group in my openldap configuration, but the redirect problem remains. Worst of all, when I enter the openDCIM URL in my browser, I should be authenticating myself as I did with basic authentication before I get the ldap login page. But after the ldap authentication, I don't have the blank page I showed before. 
![image](https://github.com/user-attachments/assets/f8cb5c3b-8ef5-4d9e-8a69-e9edd536d5b3) 
![image](https://github.com/user-attachments/assets/4422c166-e0b9-47aa-a9de-141182ee9600)

**Comment by timlegge** [comment]
Everytime I see a LDAP thread I wonder why on earth are you not using SAML
or some other SSO.  For all the complexities of SAML it is a lot easier to
set up (my opinion) than LDAP.....

Sorry, not helpful at all...

Tim

Timothy Legge
***@***.***
***@***.***

**Comment by cedgnans** [comment]
Can I have the exact meaning of every group in openDCIM? 
Is there any documentation on it?
